internal/k8s/controller.go

```diff
@@ -2277,35 +2278,48 @@ func (lbc *LoadBalancerController) createVirtualServerEx(virtualServer *conf_v1.
 	for _, u := range virtualServer.Spec.Upstreams {
 		endpointsKey := configs.GenerateEndpointsKey(virtualServer.Namespace, u.Service, u.Subselector, u.Port)
 
-		var podEndps []podEndpoint
-		var err error
+		var endps []string
+		if u.UseClusterIP {
+			s, err := lbc.getServiceForUpstream(virtualServer.Namespace, u.Service, u.Port)
+			if err != nil {
+				glog.Warningf("Error getting Service for Upstream %v: %v", u.Service, err)
+			}
+			port := strconv.Itoa(int(u.Port))
+			endps = append(endps, fmt.Sprintf("%s:%s", s.Spec.ClusterIP, port))
```

Comment:
so that there is no need in `strconv.Itoa`
(same in the VSR part)
```suggestion
			endps = append(endps, fmt.Sprintf("%s:%d", s.Spec.ClusterIP, u.Port))
```